pkg/rest/config_upload.go

```diff
@@ -194,20 +194,20 @@ func isApiDashboard(api api.Api) bool {
 
 func getExistingValuesFromEndpoint(client *http.Client, theApi api.Api, url string, apiToken string) (values []api.Value, err error) {
 
-	values = make([]api.Value, 0)
+	existingValues := make([]api.Value, 0)
 	resp, err := get(client, url, apiToken)
 
 	if err != nil {
 		return nil, err
 	}
 
 	for {
-		var objmap map[string]interface{}
 
-		err, values = unmarshalJson(theApi, err, resp, values, objmap)
+		err, values, objmap := unmarshalJson(theApi, err, resp)
 		if err != nil {
 			return values, err
```

Comment:
Do you really want to return values here in case of an error?

Reply:
Values is an empty array here and objmap is an empty map. I don't think go allows the use of `nil` here.

Reply:
Slices can be `nil`. So please return `nil` here to make it explicit that we do not return 
anything.

Reply:
Ok, good input!